packages/react-core/src/components/Alert/Alert.tsx

```diff
@@ -107,49 +136,57 @@ export const Alert: React.FunctionComponent<AlertProps> = ({
     </h4>
   );
 
-  if (disableAlert === false && timeout && timeout !== 0) {
-    setTimeout(
-      () => {
-        setDisableAlert(true);
-        if (onTimeout) {
-          onTimeout();
-        }
-      },
-      timeout === true ? 8000 : timeout
-    );
-  }
+  const myOnMouseEnter = (ev: React.MouseEvent<HTMLDivElement>) => {
+    setIsMouseOver(true);
+    setTimedOutAnimation(false);
+    onMouseEnter(ev);
+  };
 
-  if (disableAlert === false) {
-    return (
-      <div
-        {...props}
-        className={customClassName}
-        aria-label={ariaLabel}
-        {...ouiaProps}
-        {...(isLiveRegion && {
-          'aria-live': 'polite',
-          'aria-atomic': 'false'
-        })}
-      >
-        <AlertIcon variant={variant} customIcon={customIcon} />
-        {isTooltipVisible ? (
-          <Tooltip content={getHeadingContent} position={tooltipPosition}>
-            {Title}
-          </Tooltip>
-        ) : (
-          Title
-        )}
-        {actionClose && (
-          <AlertContext.Provider value={{ title, variantLabel }}>
-            <div className={css(styles.alertAction)}>{actionClose}</div>
-          </AlertContext.Provider>
-        )}
-        {children && <div className={css(styles.alertDescription)}>{children}</div>}
-        {actionLinks && <div className={css(styles.alertActionGroup)}>{actionLinks}</div>}
-      </div>
-    );
-  } else {
+  const myOnMouseLeave = (ev: React.MouseEvent<HTMLDivElement>) => {
+    setIsMouseOver(false);
+    setTimeout(() => setTimedOutAnimation(true), timeoutAnimation);
```

Comment:
This timer needs to be cleared on mouse enter and restarted on mouse leave.
You also need to cancel the timer on component unmount.

Reply:
> restarted on mouse leave
Not sure how to do this part. I don't think it needs to be restarted on mouse leave since `const dismissed = timedOut && timedOutAnimation && !isMouseOver && !containsFocus;`

Reply:
looks like you changed the implementation here, so this won't be needed now